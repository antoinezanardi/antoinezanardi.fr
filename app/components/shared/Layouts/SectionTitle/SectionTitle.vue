<template>
  <div class="h2 mb-5 text-center title">
    <WrappedFontAwesomeIcon
      id="section-title-icon"
      classes="section-icon me-2"
      :icon="icon"
      :icon-color="iconColor"
    />

    <span class="section-title-text">
      {{ title }}
    </span>
  </div>
</template>

<script setup lang="ts">
import WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";
import type { SectionTitleProps } from "~/components/shared/Layouts/SectionTitle/section-title.types";

withDefaults(defineProps<SectionTitleProps>(), { iconColor: "#000000" });
</script>