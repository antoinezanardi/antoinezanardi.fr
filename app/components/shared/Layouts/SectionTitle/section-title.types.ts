type SectionTitleProps = {
  title: string;
  icon: string;
  iconColor?: string;
};

export type { SectionTitleProps };