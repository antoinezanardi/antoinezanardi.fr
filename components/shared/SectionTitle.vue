<template>
  <div class="h2 text-center mb-5 title">
    <i class="section-icon me-2" :class="icon"/>
    <span v-html="title"/>
  </div>
</template>

<script setup lang="ts">
withDefaults(defineProps<{
  title: string;
  icon: string;
  iconColor?: string;
}>(), { iconColor: "#000000" });
</script>

<style lang="scss" scoped>
  .section-icon {
    color: v-bind(iconColor);
  }
</style>