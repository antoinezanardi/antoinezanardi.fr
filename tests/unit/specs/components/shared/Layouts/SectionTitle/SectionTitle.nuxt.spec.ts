import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import type WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { SectionTitleProps } from "~/components/shared/Layouts/SectionTitle/section-title.types";

import SectionTitle from "@/components/shared/Layouts/SectionTitle/SectionTitle.vue";

describe("Section Title Component", () => {
  const defaultProps: SectionTitleProps = {
    icon: "icon",
    title: "title",
  };
  let wrapper: ReturnType<typeof mount<typeof SectionTitle>>;

  async function mountSectionTitleComponent(options: ComponentMountingOptions<typeof SectionTitle> = {}):
  Promise<ReturnType<typeof mount<typeof SectionTitle>>> {
    return mountSuspendedComponent(SectionTitle, {
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountSectionTitleComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Icon", () => {
    it("should set icon source when rendered.", () => {
      const icon = wrapper.findComponent<typeof WrappedFontAwesomeIcon>("#section-title-icon");

      expect(icon.props("icon")).toBe("icon");
    });

    it("should set icon color to default when it is not provided in props.", () => {
      const icon = wrapper.findComponent<typeof WrappedFontAwesomeIcon>("#section-title-icon");

      expect(icon.attributes("iconcolor")).toBe("#000000");
    });

    it("should set icon color to provided color when it is provided in props.", async() => {
      wrapper = await mountSectionTitleComponent({
        props: {
          ...defaultProps,
          iconColor: "#FFFFFF",
        },
      });
      const icon = wrapper.findComponent<typeof WrappedFontAwesomeIcon>("#section-title-icon");

      expect(icon.attributes("iconcolor")).toBe("#FFFFFF");
    });
  });

  describe("Title", () => {
    it("should set title when rendered.", () => {
      const title = wrapper.find<HTMLSpanElement>(".section-title-text");

      expect(title.text()).toBe("title");
    });
  });
});