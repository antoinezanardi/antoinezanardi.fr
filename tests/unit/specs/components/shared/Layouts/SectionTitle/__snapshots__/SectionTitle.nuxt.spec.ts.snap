// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Section Title Component > should match snapshot when rendered. 1`] = `
"<div class="h2 mb-5 text-center title">
  <wrapped-font-awesome-icon-stub icon="icon" iconcolor="#000000" classes="section-icon me-2" id="section-title-icon"></wrapped-font-awesome-icon-stub><span class="section-title-text">title</span>
</div>"
`;
